core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -3262,11 +3262,21 @@ private static class CastImplementor extends AbstractRexCallImplementor {
 
     @Override Expression implementSafe(final RexToLixTranslator translator,
         final RexCall call, final List<Expression> argValueList) {
-      assert call.getOperands().size() == 1;
+      assert call.getOperands().size() <= 2;
       final RelDataType sourceType = call.getOperands().get(0).getType();
 
-      // Short-circuit if no cast is required
       RexNode arg = call.getOperands().get(0);
+      ConstantExpression formatExpr;
+      if (call.getOperands().size() > 1) {
```

Comment:
Makes sense, yes the options should only be 1 or 2 for now, if we have a `AT <Timezone>` clause added in later it would become 3, but irrespective we should make it a deterministic equality check.

Reply:
In situations like this, I use `> 1`. It is more future-proof, and asks for only what it needs.

Reply:
In the future if there was another argument we may not want it to follow the same logic, that's why I think equivalency is a better check.